src/PowerShellEditorServices/Services/PowerShellContext/Session/PSReadLinePromptContext.cs

```diff
@@ -79,23 +84,37 @@ internal class PSReadLinePromptContext : IPromptContext
         internal static bool TryGetPSReadLineProxy(
             ILogger logger,
             Runspace runspace,
-            out PSReadLineProxy readLineProxy)
+            out PSReadLineProxy readLineProxy,
+            bool testing = false)
         {
             readLineProxy = null;
             logger.LogTrace("Attempting to load PSReadLine");
             using (var pwsh = PowerShell.Create())
             {
                 pwsh.Runspace = runspace;
                 pwsh.AddCommand("Microsoft.PowerShell.Core\\Import-Module")
-                    .AddParameter("Name", _psReadLineModulePath)
+                    .AddParameter("Name", testing ? _psReadLineTestModulePath : _psReadLineModulePath)
                     .Invoke();
 
+                if (pwsh.HadErrors)
+                {
+                    logger.LogWarning("PSConsoleReadline type not found: {Reason}", pwsh.Streams.Error[0].ToString());
+                    return false;
+                }
+
                 var psReadLineType = Type.GetType("Microsoft.PowerShell.PSConsoleReadLine, Microsoft.PowerShell.PSReadLine2");
 
                 if (psReadLineType == null)
                 {
-                    logger.LogWarning("PSConsoleReadline type not found: {Reason}", pwsh.HadErrors ? pwsh.Streams.Error[0].ToString() : "<Unknown reason>");
-                    return false;
+                    logger.LogTrace("PSConsoleReadline type not found using Type.GetType(), searching all loaded assemblies...");
```

Comment:
@rjmholt This should only happen in the case that it needs to. When `Type.GetType` works, then this logic is skipped. So it feels fine, though I still wish I knew _why_ it was necessary.

Reply:
Not saying it's "right" but I keep seeing code similar to this come up when I research the issue. I think we will get to the bottom of it as we continue to refactor but this works.

Reply:
My suspicion is that this always worked in script because it always runs on the pipeline thread and through PowerShell's own type search logic. When we sift through the whole AppDomain (which is kind of a dead concept) for a type, we ignore assembly load contexts, and so that might be why the direction reflection query fails but the search succeeds. Just a theory though. An ALC shouldn't be thread-dependent though and I don't think anyone is using something like `EnterContextualReflection()`